Analysis/Tasks/PWGGA/gammaConversionsMC.cxx

```diff
@@ -0,0 +1,324 @@
+    kV0Radius,
+    kArmenteros,
+    kPsiPair,
+    kCosinePA,
+    kPhotonOut
+  };
+
+  std::vector<TString> fPhotCutsLabels{
+    "kPhotonIn",
+    "kTrackEta",
+    "kTrackPt",
+    "kElectronPID",
+    "kPionRejLowMom",
+    "kPionRejHighMom",
+    "kTPCNClsOverFindable",
+    "kV0Radius",
+    "kArmenteros",
+    "kPsiPair",
+    "kCosinePA",
+    "kPhotonOut"};
+
+  void init(InitContext const&)
+  {
+    TAxis* lXaxis = registry.get<TH1>(HIST("IsPhotonSelected"))->GetXaxis();
+    for (size_t i = 0; i < fPhotCutsLabels.size(); ++i) {
+      lXaxis->SetBinLabel(i + 1, fPhotCutsLabels[i]);
+    }
+  }
+
+  void process(aod::Collision const& theCollision,
+               aod::V0Datas const& theV0s,
+               tracksAndTPCInfo const& theTracks,
+               aod::McParticles const& theMcParticles)
+  {
+    for (auto& lV0 : theV0s) {
+
+      fillHistogramsBeforeCuts(lV0);
+
+      auto lTrackPos = lV0.template posTrack_as<tracksAndTPCInfo>(); //positive daughter
+      auto lTrackNeg = lV0.template negTrack_as<tracksAndTPCInfo>(); //negative daughter
+
+      // apply track cuts
+      float lTPCNClsOverFindablePos = -1.;
+      float lTPCNClsOverFindableNeg = -1.;
+
+      if (!(trackPassesCuts(lTrackPos, lTPCNClsOverFindablePos) && trackPassesCuts(lTrackNeg, lTPCNClsOverFindableNeg))) {
+        continue;
+      }
+
+      float lV0CosinePA = RecoDecay::CPA(array{theCollision.posX(), theCollision.posY(), theCollision.posZ()}, array{lV0.x(), lV0.y(), lV0.z()}, array{lV0.px(), lV0.py(), lV0.pz()});
+
+      // apply photon cuts
+      if (!passesPhotonCuts(lV0, lV0CosinePA)) {
+        continue;
+      }
+
+      fillHistogramsAfterCuts(lV0, lTrackPos, lTrackNeg, lTPCNClsOverFindablePos, lTPCNClsOverFindableNeg, lV0CosinePA);
+
+      processTruePhotons(lV0, lTrackPos, lTrackNeg, theMcParticles);
+    }
+  }
+
+  template <typename T>
+  void fillHistogramsBeforeCuts(const T& theV0)
+  {
+    // fill some QA histograms before any cuts
+    registry.fill(HIST("beforeCuts/hPtRec_before"), theV0.pt());
+    registry.fill(HIST("beforeCuts/hEtaRec_before"), theV0.eta());
+    registry.fill(HIST("beforeCuts/hPhiRec_before"), theV0.phi());
+    registry.fill(HIST("beforeCuts/hConvPointR_before"), theV0.v0radius());
+    registry.fill(HIST("IsPhotonSelected"), kPhotonIn);
+  }
+
+  template <typename T>
+  bool trackPassesCuts(const T& theTrack, float& theTPCNClsOverFindable)
+  {
+    // single track eta cut
+    if (TMath::Abs(theTrack.eta()) > fEtaCut) {
+      registry.fill(HIST("IsPhotonSelected"), kTrackEta);
+      return kFALSE;
+    }
+
+    // single track pt cut
+    if (theTrack.pt() < fSinglePtCut) {
+      registry.fill(HIST("IsPhotonSelected"), kTrackPt);
+      return kFALSE;
+    }
+
+    if (!(selectionPIDTPC_track(theTrack))) {
+      return kFALSE;
+    }
+
+    theTPCNClsOverFindable = (float)theTrack.tpcNClsFound() / (float)theTrack.tpcNClsFindable();
+    if (theTPCNClsOverFindable < fMinTPCNClsOverFindable) {
```

Comment:
There is a dynamic column TPCCrossedRowsOverFindableCls. Here you use found clusters over findable.
Two questions:
- isn't the cut with crossed rows the better quantity?
- if not, would it help to add a dynamic column TPCFoundOverFindableCls? In this way you don't have to pass the value by reference out of the function.

Reply:
q1) I used the found/findable since this is what we are using in gamma conversion code in AliPhysics. But so now I also added TPCCrossedRowsOverFindableCls so we can use that one if it's better
q2) yes, I added it now, definitely the cleaner solution